lib/resty/core/base.lua

```diff
@@ -69,14 +74,38 @@ end
 do
     local orig_require = require
     local pkg_loaded = package.loaded
-    local function my_require(name)
-        local mod = pkg_loaded[name]
-        if mod then
-            return mod
+    local key_sentinel
+
+    local function get_key_sentinel()
+        local n = C.lj_ffi_get_key_sentinel()
+        key_sentinel = tonumber(n)
+    end
+
+    pcall(get_key_sentinel)
+
+    if key_sentinel ~= nil then
+        local function my_require(name)
+            local mod = pkg_loaded[name]
+            if mod then
+                if type(mod) == 'number' and mod == key_sentinel then
```

Comment:
Is the `type(mod) == 'number'` check required. I do not see how `mod == key_sentinel` could invoke the `__eq` metamethod.